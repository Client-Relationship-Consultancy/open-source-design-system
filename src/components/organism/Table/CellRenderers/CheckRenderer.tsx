import React from "react";
import styled from "styled-components";
import Icon from "../../../atoms/Icon";

interface IIconWrapper {
  size?: string;
}

export const IconWrapper = styled.div<IIconWrapper>`
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: ${(props): string => props.size || "1.5rem"};
`;
IconWrapper.displayName = "IconWrapper";

interface IProps {
  value: boolean;
  size?: string;
}

const CheckRenderer: React.FC<IProps> = ({ value, size }: IProps) => {
  if (value === true || value === false) {
    return (
      <IconWrapper size={size}>
        {value === true ? (
          <Icon name="check" color="primary" />
        ) : (
          <Icon name="times" color="error" />
        )}
      </IconWrapper>
    );
  }
  return null;
};

export default CheckRenderer;
