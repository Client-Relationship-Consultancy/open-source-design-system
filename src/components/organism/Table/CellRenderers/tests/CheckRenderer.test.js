import React from "react"
import { shallow, mount } from "enzyme"
import CheckRenderer, { IconWrapper } from "../CheckRenderer"
import "jest-styled-components"

describe("<CheckRenderer />", () => {
  it("should match snapshot", () => {
    const component = shallow(<CheckRenderer value />)
    expect(component).toMatchSnapshot()
  })

  it("should render a check if the value is true", () => {
    const component = shallow(<CheckRenderer value />)
    const iconProps = component.find("Icon").props()
    expect(iconProps.name).toEqual("check")
    expect(iconProps.color).toEqual("primary")
  })

  it("should render a cross/times if the value is false", () => {
    const component = shallow(<CheckRenderer value={false} />)
    const iconProps = component.find("Icon").props()
    expect(iconProps.name).toEqual("times")
    expect(iconProps.color).toEqual("error")
  })

  it("should render nothing if the value is not a true or false", () => {
    const component = shallow(<CheckRenderer value={null} />)
    expect(component.find("Icon").exists()).toEqual(false)

    component.setProps({ value: "alright" })
    expect(component.find("Icon").exists()).toEqual(false)

    component.setProps({ value: "" })
    expect(component.find("Icon").exists()).toEqual(false)

    component.setProps({ value: undefined })
    expect(component.find("Icon").exists()).toEqual(false)

    component.setProps({ value: 123 })
    expect(component.find("Icon").exists()).toEqual(false)
  })

  it("should pass a custom size to the wrapper when a size is given", () => {
    const component = shallow(<CheckRenderer value size="20px" />)
    expect(component.find("IconWrapper").props().size).toEqual("20px")
  })
})

describe("<IconWrapper />", () => {
  it("should match snapshot", () => {
    const component = mount(<IconWrapper />)
    expect(component).toMatchSnapshot()
  })

  it("should default to 1.5rem when no size is given", () => {
    const component = mount(<IconWrapper />)
    expect(component).toHaveStyleRule("font-size", "1.5rem")
  })

  it("should use the custom size if size is provided", () => {
    const component = mount(<IconWrapper size="100px" />)
    expect(component).toHaveStyleRule("font-size", "100px")
  })
})
