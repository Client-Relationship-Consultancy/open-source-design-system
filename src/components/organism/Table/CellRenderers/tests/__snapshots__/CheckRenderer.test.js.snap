// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckRenderer /> should match snapshot 1`] = `
<IconWrapper>
  <Icon
    color="primary"
    name="check"
  />
</IconWrapper>
`;

exports[`<IconWrapper /> should match snapshot 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1.5rem;
}

<IconWrapper>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "CheckRenderer__IconWrapper-sc-1wvjnyh-0",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "width:100%;display:flex;justify-content:center;align-items:center;font-size:",
            [Function],
            ";",
          ],
        },
        "displayName": "IconWrapper",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "CheckRenderer__IconWrapper-sc-1wvjnyh-0",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="c0"
    />
  </StyledComponent>
</IconWrapper>
`;
